src/main/java/duke/tasks/Deadline.java

```diff
@@ -0,0 +1,46 @@
+package duke.tasks;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Deadline is a class which represents the deadline task.
+ */
+public class Deadline extends Task {
+    protected LocalDate by;
```

Comment:
The name of the variable 'by' could perhaps be changed to something more easily identifiable for other readers? Maybe 'date' or 'deadlineDate'?